src/pages/organisationUnitLevels/ListRows.tsx

```diff
@@ -0,0 +1,251 @@
+import { useAlert, useDataEngine } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import {
+    DataTableCell,
+    DataTableRow,
+    Input,
+    SingleSelect,
+    SingleSelectOption,
+} from '@dhis2/ui'
+import { useQuery } from '@tanstack/react-query'
+import React, { useCallback, useState } from 'react'
+import { SectionList } from '../../components'
+import { useModelListView } from '../../components/sectionList/listView'
+import { ModelValue } from '../../components/sectionList/modelValue/ModelValue'
+import css from '../../components/sectionList/SectionList.module.css'
+import { SectionListLoader } from '../../components/sectionList/SectionListLoader'
+import {
+    SectionListEmpty,
+    SectionListError,
+} from '../../components/sectionList/SectionListMessages'
+import {
+    BaseListModel,
+    DEFAULT_FIELD_FILTERS,
+    getViewConfigForSection,
+    SECTIONS_MAP,
+    useSchemaFromHandle,
+} from '../../lib'
+import { useOnEditCompletedSuccessfully } from '../../lib/form/useOnSubmit'
+import { getFieldFilter } from '../../lib/models/path'
+import { Access } from '../../types/generated'
+import { EditingListActions } from './EditingListActions'
+import { ViewingListActions } from './ViewingListActions'
+
+export type OrgUnitListModel = {
+    id: string
+    name: string
+    access: Access
+    displayName: string
+    offlineLevels?: number
+    index?: number
+}
+type ModelListResponse = { result: OrgUnitListModel[] }
+
+export const ListRows = ({
+    onTranslationClick,
+}: {
+    onTranslationClick: (model: OrgUnitListModel) => void
+}) => {
+    const schema = useSchemaFromHandle()
+    const { columns: headerColumns } = useModelListView()
+    const dataEngine = useDataEngine()
+    const onEditCompletedSuccessfully = useOnEditCompletedSuccessfully(
+        SECTIONS_MAP.organisationUnitLevel
+    )
+    const saveAlert = useAlert(
+        ({ message }) => message,
+        (options) => options
+    )
+    const [editingOrgUnitLevel, setEditingOrgUnitLevel] = useState<
+        undefined | OrgUnitListModel
+    >(undefined)
+
+    const query = {
+        result: {
+            resource: 'filledOrganisationUnitLevels',
+            params: {
+                order: 'level',
+                fields: getViewConfigForSection(schema.name)
+                    .columns.available.map((column) =>
+                        getFieldFilter(schema, column.path)
+                    )
+                    .concat(DEFAULT_FIELD_FILTERS)
+                    .concat(['name']),
+            },
+        },
+    }
+
+    const { error, data, refetch } = useQuery({
+        queryKey: [query],
+        queryFn: ({ queryKey: [query], signal }) => {
+            return dataEngine.query(query, {
+                signal,
+            }) as Promise<ModelListResponse>
+        },
+    })
```

Comment:
Lets try to follow the same pattern as most of the app. This will bind the queryFn give correct type etc. 
```suggestion
      const queryFn = useBoundResourceQueryFn()
     const { error, data, refetch } = useQuery({
        queryKey: [
            {
                resource: 'filledOrganisationUnitLevels',
                params: {
                    order: 'level',
                    fields: getViewConfigForSection(schema.name)
                        .columns.available.map((column) =>
                            getFieldFilter(schema, column.path)
                        )
                        .concat(DEFAULT_FIELD_FILTERS)
                        .concat(['name']),
                },
            },
        ],
        queryFn: queryFn<ModelListResponse>,
    })
```